modules/validations/outdated_tags.js

```diff
@@ -93,63 +160,120 @@ export function validationOutdatedTags() {
+        if (k === 'building') {
+          v = 'yes';
+          if (preset.id !== 'building/yes') continue;  // the feature matched a better preset
         }
-        // I considered setting `name` and other tags here, but they aren't unique per wikidata
-        // (Q2759586 -> in USA "Papa John's", in Russia "Папа Джонс")
-        // So users will really need to use a preset or assign `name` themselves.
-      }
 
-      // try key/value|name match against name-suggestion-index
-      if (newTags.name) {
-        for (let i = 0; i < nsiKeys.length; i++) {
-          const k = nsiKeys[i];
-          if (!newTags[k]) continue;
-
-          const center = entity.extent(graph).center();
-          const countryCode = countryCoder.iso1A2Code(center);
-          const match = _nsi.matcher.matchKVN(k, newTags[k], newTags.name, countryCode && countryCode.toLowerCase());
-          if (!match) continue;
-
-          // for now skip ambiguous matches (like Target~(USA) vs Target~(Australia))
-          if (match.d) continue;
-
-          const brand = _nsi.brands[match.kvnd];
-          if (brand && brand.tags['brand:wikidata'] &&
-            brand.tags['brand:wikidata'] !== entity.tags['not:brand:wikidata']) {
-            subtype = 'noncanonical_brand';
-
-            const keepTags = ['takeaway'].reduce((acc, k) => {
-              if (newTags[k]) {
-                acc[k] = newTags[k];
-              }
+        if (!loc) {     // collect location for this feature only once
+          loc = entity.extent(graph).center();
+        }
+        if (!names) {   // collect names for this feature only once
+          names = Object.keys(newTags)
+            .map(k => isNamelike(k) ? newTags[k] : null)
+            .filter(Boolean);
+
+          if (foundQID) names.unshift(foundQID);  // matcher will recognize the QID as a name too
+          names = utilArrayUniq(names);
+        }
+
+        // Try each namelike value
+        for (let n = 0; n < names.length; n++) {
+          match = _nsi.matcher.match(k, v, names[n], loc);   // Attempt to match an item in NSI
+          if (!match) continue;  // keep looking
+
+          // If we get here, there was a match..
+          // A match may contain multiple results, the first one is the best one for this location
+          // e.g. `['pfk-a54c14', 'kfc-1ff19c', 'kfc-658eea']`
+          const itemID = match[0].itemID;
+          const item = _nsi.ids.get(itemID);
+          const mainTag = item.mainTag;               // e.g. `brand:wikidata`
+          const itemQID = item.tags[mainTag];         // e.g. `brand:wikidata` qid
+          const notQID = newTags[`not:${mainTag}`];   // e.g. `not:brand:wikidata` qid
+
+          // Exceptions, throw out the match
+          if (
+            (!itemQID || itemQID === notQID) ||       // no `*:wikidata` or matched a `not:*:wikidata`
+            (newTags.office && !item.tags.office)     // feature may be a coprorate office for a brand? - #6416
```

Comment:
Would a corporate office still be a decent candidate for an `operator` preset?

Reply:
I'm not sure.. I kind of think no, except in cases where it's really widespread or like a chain.  (Military recruiting centers maybe or DMVs?)

I suppose any business can also have a corporate office.  Do people tag them like `office=yes`?  Adding an extra preset for each one seems like it would just confuse users.

For reference, these are the brand-like `office=*` categories NSI tracks today:
<img width="300" alt="Screen Shot 2021-01-18 at 7 17 31 PM" src="https://user-images.githubusercontent.com/38784/104973139-da3e2c80-59c1-11eb-9ebd-9130663c3152.png">